rpc/client.go

```diff
@@ -325,7 +326,16 @@ func (c *Client) runHeartbeat(retryOpts retry.Options) {
 				}
 				if err = c.connect(); err != nil {
 					setUnhealthy()
-					log.Warning(err)
+					switch err.(type) {
+					case x509.HostnameError, x509.CertificateInvalidError, x509.UnknownAuthorityError,
+						x509.SystemRootsError, x509.InsecureAlgorithmError, x509.ConstraintViolationError,
+						x509.UnhandledCriticalExtension:
```

Comment:
This set of `x509` errors might be overly broad. The one we see commonly is `x509.HostnameError` which is returned when the hostname we're connecting to is not listed in the certificate.